code/game/atoms.dm

```diff
@@ -217,9 +218,15 @@ var/global/list/ghdel_profiling = list()
 	if (density == src.density)
 		return FALSE // No need to invoke the event when we're not doing any actual change
 	src.density = density
+	densityChanged()
+
+/atom/proc/densityChanged()
 	INVOKE_EVENT(on_density_change, list("atom" = src)) // Invoke event for density change
```

Comment:
couldn't you just subscribe to this event?

Reply:
That's for the object's density change, not the turfs contents density change.

Reply:
And the reason I'm not putting the on_turf_density_change even on the turf itself, is because we've got 1500000 turfs, but only about 100~ areas